### DISCUSSION TITLE
Error: addDoubletScores about ggplot2

### DISCUSSION THREAD

**Discussion by yuyuyuto001** [GreenleafLab/ArchR#2124] [Questions / Documentation]
Hey everybody.

I had an error when I used addDoubletScores function.

**Attach log files
ArchR has built-in logging capabilities for complex functions. A log file (which will appear in the console output) must be attached to this issue. Drag and drop it here.
[ArchR-addDoubletScores-1b70395bc576-Date-2024-02-29_Time-09-57-38.155893.log](https://github.com/GreenleafLab/ArchR/files/14441560/ArchR-addDoubletScores-1b70395bc576-Date-2024-02-29_Time-09-57-38.155893.log)

Bug Description.
When I use the addDoubletScores function, I get the following error.

> > doubScores <- addDoubletScores(
> +   input = ArrowFiles,
> +   k = 10, #Refers to how many cells near a "pseudo-doublet" to count.
> +   knnMethod = "UMAP", #Refers to the embedding to use for nearest neighbor search.
> +   LSIMethod = 1
> + )
> ArchR logging to : ArchRLogs/ArchR-addDoubletScores-1b70395bc576-Date-2024-02-29_Time-09-57-38.155893.log
> If there is an issue, please report to github with logFile!
> 2024-02-29 09:57:38.221589 : Batch Execution w/ safelapply!, 0 mins elapsed.
> 2024-02-29 09:57:38.231848 : scATAC_BMMC_R1 (1 of 3) :  Computing Doublet Statistics, 0 mins elapsed.
> Warning: Data is of class matrix. Coercing to dgCMatrix.
> Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'
> Also defined by ‘spam’
> Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'
> Also defined by ‘spam’
> scATAC_BMMC_R1 (1 of 3) : UMAP Projection R^2 = 0.97882
> 
> ************************************************************
> 2024-02-29 10:00:00.474936 : ERROR Found in ggplot for scATAC_BMMC_R1 (1 of 3) :  
> LogFile = ArchRLogs/ArchR-addDoubletScores-1b70395bc576-Date-2024-02-29_Time-09-57-38.155893.log
> 
> <simpleError in g$grobs[[legend]]: no such index at level 2
> >
> 
> ************************************************************
> 
> 2024-02-29 10:00:00.507234 : scATAC_CD34_BMMC_R1 (2 of 3) :  Computing Doublet Statistics, 2.371 mins elapsed.
> Warning: Data is of class matrix. Coercing to dgCMatrix.
> Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'
> Also defined by ‘spam’
> Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'
> Also defined by ‘spam’
> scATAC_CD34_BMMC_R1 (2 of 3) : UMAP Projection R^2 = 0.99203
> 
> ************************************************************
> 2024-02-29 10:01:55.924854 : ERROR Found in ggplot for scATAC_CD34_BMMC_R1 (2 of 3) :  
> LogFile = ArchRLogs/ArchR-addDoubletScores-1b70395bc576-Date-2024-02-29_Time-09-57-38.155893.log
> 
> <simpleError in g$grobs[[legend]]: no such index at level 2
> >
> 
> ************************************************************
> 
> 2024-02-29 10:01:55.957149 : scATAC_PBMC_R1 (3 of 3) :  Computing Doublet Statistics, 4.296 mins elapsed.
> Warning: Data is of class matrix. Coercing to dgCMatrix.
> Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'
> Also defined by ‘spam’
> Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'
> Also defined by ‘spam’
> scATAC_PBMC_R1 (3 of 3) : UMAP Projection R^2 = 0.96012
> 
> ************************************************************
> 2024-02-29 10:03:56.628193 : ERROR Found in ggplot for scATAC_PBMC_R1 (3 of 3) :  
> LogFile = ArchRLogs/ArchR-addDoubletScores-1b70395bc576-Date-2024-02-29_Time-09-57-38.155893.log
> 
> <simpleError in g$grobs[[legend]]: no such index at level 2
> >
> 
> ************************************************************
> 
> ArchR logging successful to : ArchRLogs/ArchR-addDoubletScores-1b70395bc576-Date-2024-02-29_Time-09-57-38.155893.log

**How to reproduce
I can reproduce it with the tutorial dataset, and the above code is tutorial's.

Additional Context.
I reinstalled ArchR and ggplot2, and tried to restart R and R studio several times, but the problem wasn't solved.
Could you help me to solve this error?
Thanks in advance!

**Comment by huiqizhu** [comment]
I have the same error. Have you solved this?

**Reply by nansne** [reply]
I encountered this error during call peaking. The call peaking process has been completed, However, I got an error message that says "Error in g$grobs[[legend]]: no such index at level 2". Have you solved this?

**Comment by nansne** [comment]
I encountered this error during call peaking. The call peaking process has been completed, However, I got an error message that says "Error in g$grobs[[legend]]: no such index at level 2". Have you solved this?

**Reply by nansne** [reply]
ggplot2 version changed to 3.4.4,not 3.5

**Comment by alekseybelikov** [comment]
solved ggplot errors by downgrading to version 3.4.0

```
ggplot_url <- "https://cran.r-project.org/src/contrib/Archive/ggplot2/ggplot2_3.4.0.tar.gz"
install.packages(ggplot_url, repos=NULL, type="source")
```
